### PR TITLE
13 sonarcloud automatic code review

### LINKED ISSUE #13: SonarCloud automatic code review
Automation is good. So lets add scanning of our code with [SonarCloud](https://sonarcloud.io/). It will check our code for Code Smells, Bugs and Vulnerabilities. 

Can be triggered by [sonarcloud-github-action](https://github.com/SonarSource/sonarcloud-github-action)

### PR DESCRIPTION
**Pull request by ArmanAlemirad**
Adding sonarCloud. 
Fixes #13

### CODE REVIEW

File: .github/workflows/sonar.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: mvn -B verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.projectKey=fungover_haze
```
Comment (safstromo) [inline .github/workflows/sonar.yml:32]:
```suggestion
        run: mvn -B verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.projectKey=fungover_haze

```
Add new line at the end of file 😀💪
Other than that all good

**Review by safstromo — CHANGES_REQUESTED** [review]

**Review by RobertMili — APPROVED** [review]
This look good for me 👍

**Review by RobertMili — DISMISSED** [review]
Looks good

**Comment by wassimtajeddin** [comment]
It seems good

**Review by wassimtajeddin — APPROVED** [review]

**Review by wassimtajeddin — APPROVED** [review]

**Review by wassimtajeddin — APPROVED** [review]

**Review by wassimtajeddin — APPROVED** [review]

**Review by wassimtajeddin — APPROVED** [review]

**Review by wassimtajeddin — APPROVED** [review]

**Pull request merged**